src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/EventListener.java

```diff
@@ -164,25 +166,39 @@ protected void schedule(GerritTrigger t, GerritCause cause, GerritTriggeredEvent
      * @param event   the event.
      * @param project the project to build.
      */
-    protected void schedule(GerritTrigger t, GerritCause cause, GerritTriggeredEvent event, AbstractProject project) {
+    protected void schedule(GerritTrigger t, GerritCause cause, GerritTriggeredEvent event, Job project) {
         BadgeAction badgeAction = new BadgeAction(event);
         //during low traffic we still don't want to spam Gerrit, 3 is a nice number, isn't it?
         int projectbuildDelay = t.getBuildScheduleDelay();
         if (cause instanceof GerritUserCause) {
             // it's a manual trigger, no need for a quiet period
             projectbuildDelay = 0;
-        } else if (project.getHasCustomQuietPeriod()
-                && project.getQuietPeriod() > projectbuildDelay) {
-            projectbuildDelay = project.getQuietPeriod();
+        } else if (project instanceof AbstractProject) {
+            AbstractProject abstractProject = (AbstractProject)project;
+            if (abstractProject.getHasCustomQuietPeriod() && abstractProject.getQuietPeriod() > projectbuildDelay) {
+                projectbuildDelay = abstractProject.getQuietPeriod();
+            }
         }
         ParametersAction parameters = createParameters(event, project);
-        Future build = project.scheduleBuild2(
+
+        Future build;
+        if (project instanceof AbstractProject) {
+            build = ((AbstractProject)project).scheduleBuild2(
                 projectbuildDelay,
                 cause,
                 badgeAction,
                 new RetriggerAction(cause.getContext()),
                 new RetriggerAllAction(cause.getContext()),
                 parameters);
+        } else if (project instanceof SCMTriggerItem) {
+            // TODO: Using SCMTriggerItem smells bad here. What should we be using?
```

Comment:
This causes a regression for Workflows because we don't actually include the parameters.

We can pass all the actions, and at minimum we should do that even if we stick to SCMTriggerItem. You failed to pass badgeAction, and both Retrigger actions, as well as the parameters.

We can't pass the cause directly though, at least not to SCMTriggerItem